viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
+from ast import And
+from itertools import count
+from site import USER_BASE
+from typing import Counter
+import copy # I want to import copy module to use deepcopy
+'''I have chosen to use the deepcopy to ensure that I am not making
+changes to the original list pass through the functions.
+I am making changes to the new list for three(3) of the functions in this wave.
+However, as we know, if I only make a shallow copy, it will also change the 
+original list since the elements in both the original list and the new list
+are referencing the same ID number. By creating a deepcopy, I can ensure that as 
+the function iterates through the old list and changes to the new list,  the 
+changes will only affect the new list.'''
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    if new_movie["title"] == None or\
+        new_movie["genre"] == None or\
+            new_movie["rating"] == None:
+        return None     
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    watched_movie = copy.deepcopy(user_data)
+    watched_movie["watched"].append(movie)
+    return watched_movie
+
+def add_to_watchlist(user_data, movie):
+    watchlist_movie = copy.deepcopy(user_data)
+    watchlist_movie["watchlist"].append(movie)
+    return watchlist_movie
+
+def watch_movie(user_data, movie_title):
+    updated_movie = copy.deepcopy(user_data)
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            updated_movie["watched"].append(movie)
+            updated_movie["watchlist"].remove(movie)
+    return updated_movie
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    my_ratings_total = 0
+    movie_len = len(user_data["watched"])
+    if not user_data["watched"]:
+        return 0
+    

```

Comment:
Let's always bring the guard clause all the way up to the top of the function before we start making variables.

```suggestion
    if not user_data["watched"]:
        return 0
    my_ratings_total = 0
    movie_len = len(user_data["watched"])
```